Codec/Archive/Tar/Check/Internal.hs

```diff
@@ -65,12 +66,17 @@ checkSecurity = checkEntries checkEntrySecurity
 
 -- | @since 0.6.0.0
 checkEntrySecurity :: Entry -> Maybe FileNameError
-checkEntrySecurity entry = case entryContent entry of
-    HardLink     link -> check (entryPath entry)
-                 `mplus` check (fromLinkTarget link)
-    SymbolicLink link -> check (entryPath entry)
-                 `mplus` check (fromLinkTarget link)
-    _                 -> check (entryPath entry)
+checkEntrySecurity entry = check (entryPath entry) <|> case entryContent entry of
+    HardLink     link -> check (fromLinkTarget link)
+    SymbolicLink link -> check (FilePath.Native.takeDirectory (entryPath entry)
+                                FilePath.Native.</> fromLinkTarget link)
```

Comment:
Why `Native`? `tar` expects unix style separators ("slash" in the spec text) for its entries and so we should also expect these semantics for symlink target.

Now I also realize we're quite sloppy about this fact in `unpack` and `pack`. I think a symlink created on windows might not survive unpack on unix.

I'll probably have to test that.

Reply:
Yes, I tested it. Symlinks are broken when packed on windows and trying to unpack on unix. We have to splitPath and joinPath them.